modules/apps/object/object-rest-impl/src/main/java/com/liferay/object/rest/internal/odata/filter/expression/PredicateExpressionVisitorImpl.java

```diff
@@ -662,7 +662,17 @@ private Object _getValue(
 
 				Date date = dateFormat.parse(value);
 
-				right = format.format(date);
+				if (Objects.equals(
+						DBManagerUtil.getDBType(), DBType.POSTGRESQL)) {
+
+					right = new Timestamp(date.getTime());
```

Comment:
Why are we creating a Timestamp? is it not enough with Date?

Reply:
Or with a long value?

Reply:
when with Postgres we receive an error:
`org.postgresql.util.PSQLException: ERROR: operator does not exist: timestamp without time zone >= character varying_  Hint: No operator matches the given name and argument types. You might need to add explicit type casts._ `

Reply:
I tested some days ago and I think it works with using Date instead of Timestamp. Have you tried?

Reply:
I haven't. Why? Cause I didn't understand what was happening in the code 😆 If I realized earlier that the date is being formatted, and doesn't need to be turned into SimpleDate for PostgreSQL... Instead, I tried the Timestamp, led by the error suggestion, and since it worked I stopped looking for other solutions. I have pushed the changes, hopefully, there's no more, as some would say "absurd" code in the solution :D